### DISCUSSION TITLE
Integrate OAuth2 Flow into a backend

### DISCUSSION THREAD

**Discussion by Fischer96** [pocketbase/pocketbase#2023] [Q&A]
Hey, i'm using Pocketbase for a while now and everything is super smooth, thanks for that!

**Just one question:** Is it possible to abstract the OAuth2 Flow in a backend? Currently i'm having the whole login inside of a React app. So the login button gets the auth provider list, navigates me to Spotify/Discord login screen and then redirects the user to the frontend again where the `authWithOAuth2` call happens.

Is it possible to abstract this whole process into a Express or NestJS backend? Let's say i want the user to login so he calls my login endpoint on the backend where the Authprovider will be extracted and the user is then redirected to the provider login view.

After that we redirect to another backend url to handle the `authWithOAuth2` and afterwards a redirect to the frontend. I can't quite get my hand around this and also how to check whether a specific user is authenticated. I want to do this so that i can easily swap out parts later on and keep the Frontend super simple. Is this possible?

Edit:

Planned flow:

- Frontend login button redirects to login endpoint on NestJS
- Provider is fetched and the user is redirected to Discord login
- Discord login redirects to redirect endpoint on backend
- OAuth2 login happens
- What now? This is where i don't really know how to handle everything

And thanks for moving the issue to a discussion, sorry for that!

**Comment by ganigeorgiev** [comment]
You need some sort of session or persistent connection to handle the flow entirely in the backend. This is because after Discord redirects back to your node action and you perform the `authWithOAuth2()` call, you need to somehow return the auth data to the user (eg. redirect with a cookie but then since the redirect is initiated by discord I'm not sure if this will work on mobile platforms).

The universal approach would be to have a persistent connection (similar flow will be handled in #55).

**Comment by Fischer96** [comment]
Okay i understand. Would be for now possible to handle the whole login still on the frontend but get data from a collection via backend?

So:

- User does the whole login on frontend
- User wants to fetch data from a collection by calling my backend
- Backend checks whether this user is authenticated and proceeds to fetch data from pocketbase and passes that back to the frontend
